app/scripts/services/globalPromotionService.ts

```diff
@@ -6,30 +6,85 @@ import { GlobalPromotion } from 'globalPromotion';
 const mockUrl = 'http://localhost:9001';
 
 export class GlobalPromotionService {
+  private globalPromosCache: {
```

Comment:
Is this codebase new enough to let us use `Map`? There's also https://docs.angularjs.org/api/ng/type/$cacheFactory.Cache, but I'd prefer `Map` if it works.